mempool/src/mempool.rs

```diff
@@ -690,6 +693,21 @@ impl MempoolState {
 
         Some(tx)
     }
+
+    // Removes all the transactions that belong to some specific sender
+    pub(crate) fn remove_sender_txns(&mut self, sender_address: &Address) {
+        //Remove all transactions from internal ordered data structures
+        for tx_hash in &self.state_by_sender.remove(sender_address).unwrap().txns {
+            self.best_transactions.remove(tx_hash);
+            self.worst_transactions.remove(tx_hash);
+            self.oldest_transactions.remove(tx_hash);
+
+            self.transactions.remove(tx_hash);
+        }
+
+        //Finally remove all the state associated with this sender
+        self.state_by_sender.remove(&sender_address);
```

Comment:
Isn't the state already removed above in line 700 when creating the loop range? Plus, there's a clippy warning on this line.